### PR TITLE
SALTO-1427 added list command to cli | changed multi-env-source getElementIdsByS…

### PR DESCRIPTION
**Pull request by emaohi**
Added new "list" sub-command to element command, to show elements by given selectors, with `env` / `common` / `both` choices. Added support to multi-envs' `getElementIdsBySelectors`.

---
_Release Notes_: 
*CLI*
Added `list` sub-command to the element command (`element list`) to list elements by selectors. supports 3 modes - `env` for environment only, `common` for common only and `both` (default) for both. environment is picked by the standard `-e` flag.

### CODE REVIEW

File: packages/cli/src/commands/element.ts [hunk 1]
```diff
@@ -465,6 +465,82 @@ const elementOpenDef = createWorkspaceCommand({
...
+        required: true,
+      },
+    ],
+    keyedOptions: [
+      ENVIRONMENT_OPTION,
+      {
+        name: 'mode',
+        alias: 'm',
+        required: false,
+        description: 'Choose a list mode. Options - [regular, common, env]',
```
Comment (netama) [inline packages/cli/src/commands/element.ts:534]: did `regular` change to `both` in one of the iterations? :) also surprising these aren't auto-generated, but I see they're this way in other commands as well. @adamro guess that's a known limitation?

File: packages/cli/src/prompts.ts [hunk 2]
```diff
@@ -257,6 +257,14 @@ ${Prompts.LIST_IDS(ids)}
   public static readonly MOVE_START = (
     to: string,
   ): string => `Moving the specified elements to ${to}.
+`
+
+  public static readonly LIST_MESSAGE = (
+    ids: readonly string[]
+  ): string => `The following configuration elements met your selector(s):
```
Comment (netama) [inline packages/cli/src/prompts.ts:264]: nit: I'm not sure about the "met your selector(s)" part - maybe "match the search criteria" or "were found"? @adamro @tomermevorach might have better suggestions

File: packages/workspace/src/workspace/nacl_files/multi_env/multi_env_source.ts [hunk 3]
```diff
@@ -316,12 +318,28 @@ const buildMultiEnvSource = (
...
+      case 'env': {
+        relevantSource = primarySource()
+        break
+      }
+      case 'common': {
+        relevantSource = commonSource()
+        break
+      }
+      default: relevantSource = await getElementsSource()
+    }
```
Comment (netama) [inline packages/workspace/src/workspace/nacl_files/multi_env/multi_env_source.ts:341]: nit: you can turn this into a function and then `relevantSource` can remain `const` (and you can also avoid the `breaks`)
```suggestion
    const determineSource = async (): Promise<ElementsSource> => {
      switch (fromSource) {
        case 'env': {
          return primarySource()
        }
        case 'common': {
          return commonSource()
        }
        default: {
          return getElementsSource()
        }
      }
    }
    const relevantSource = await determineSource()
```

**Review by netama — APPROVED** [review]
looks great :) added a few nits. 
please make sure either @adamro or @tomermevorach review the command's args before merging

File: packages/cli/src/prompts.ts (hunk 2 above)
Comment (tomermevorach) [reply packages/cli/src/prompts.ts]: How about "The following matching configuration elements were found"

File: packages/cli/src/commands/element.ts (hunk 1 above)
Comment (tomermevorach) [reply packages/cli/src/commands/element.ts]: I'd rename to regular to all

**Pull request merged**